src/app/pages/solicitudesgiros/components/table-listasolicitudes/table-listasolicitudes.component.ts

```diff
@@ -1,44 +1,123 @@
+    private store: Store<any>,
+    private router: Router,
+    private translate: TranslateService,
+    private activatedRoute: ActivatedRoute,
+    private datePipe: DatePipe
   ) {
-    this.datosTabla = DATOS_TABLAREGISTROS;
-    this.configuracion = CONFIGURACION_TABLAREGISTROS;
+    this.tituloAccion = this.activatedRoute.snapshot.url[0].path;
+    this.datosTabla = [];
+    this.configSolicitudes = CONFIGURACION_TABLAREGISTROS;
+    for (let i = 0; i < this.configSolicitudes.dataConfig; i++) {
+      this.configSolicitudes.dataConfig[i].title.name = this.translate.instant('BANCO.' + this.configSolicitudes.dataConfig[i].title.name_i18n);
+    }
 
     this.stringBusqueda = '';
     this.selectedAction = new EventEmitter<any>();
+    this.store.dispatch(getSolicitudesGiro({sortby: ['Numero_Solicitud'], order: ['desc']}));
+  }
+  ngOnDestroy(): void {
+    this.datosTabla = [];
   }
 
   ngOnInit() {
-    this.subscription$ = this.store.select(getFilaSeleccionada).subscribe((fila: any) => {
+    this.displayedColumns = this.configSolicitudes.dataConfig.map(x => x.key);
+    this.columnNames = this.configSolicitudes.dataConfig;
+    this.subSolicitudesGiro$ = this.store.select(getSolicitudesGiro).subscribe((accion: any) => {
+      if (accion && accion.solicitudesgiros.SolicitudesGiro) {
+        this.solicitudesGiro = accion.solicitudesgiros.SolicitudesGiro.SolicitudesGiro;
+        this.buildTable();
+      }
+    });
 
+    this.subscription$ = this.store.select(getFilaSeleccionada).subscribe((fila: any) => {
       if (fila) {
-
         this.store.dispatch(loadSolicitudgiroSeleccionado(fila.fila));
       }
     });
-    this.subscription$ = this.store.select(getAccionTabla).subscribe((accion: any) => {
 
+    this.subscription$ = this.store.select(getAccionTabla).subscribe((accion: any) => {
       this.store.dispatch(loadSolicitudgiroSeleccionado(null));
     });
   }
+
+  enviarRevision(solicitud: any) {
+    const element = this.solicitudesGiro[this.solicitudesGiro.findIndex((e: any) => e.Numero_Solicitud === solicitud.NumeroSolicitud)];
+    element.Estado = 'Por revisar';
+    this.store.dispatch(actualizarAutorizacionGiro({id: element._id, element: element, path: this.tituloAccion}));
+  }
+
+  revision(solicitud: any) {
+    this.router.navigateByUrl('pages/solicitudesgiros/revisar/' + solicitud.Id);
+  }
+
+  buildTable() {
+    const tableArr: Element[] = [];
+    for (let index = 0; index < this.solicitudesGiro.length; index++) {
+      const element: Element = {
+        NumeroSolicitud: this.solicitudesGiro[index].Numero_Solicitud,
+        NombreBeneficiario: this.solicitudesGiro[index].Nombre_Beneficiario,
+        Fecha: this.datePipe.transform(this.solicitudesGiro[index].Fecha_creacion, 'dd-MM-yyyy'),
+        Estado: this.solicitudesGiro[index].Estado,
+        estado: true,
+        Id: this.solicitudesGiro[index]._id,
+        acciones: ''
+      };
+      tableArr.push(element);
+    }
```

Comment:
En general/En lo posible, [evitar los for](https://blog.sapegin.me/all/avoid-loops/) y usar métodos de array
```suggestion
    const tableArr: Element[] = this.solicitudesGiro.map(solicitud => { return <Element>{
      NumeroSolicitud: solicitud.Numero_Solicitud,
      NombreBeneficiario: solicitud.Nombre_Beneficiario,
      Fecha: this.datePipe.transform(solicitud.Fecha_creacion, 'dd-MM-yyyy'),
      Estado: solicitud.Estado,
      estado: true,
      Id: solicitud._id,
      acciones: '',
    }});
```